### PR TITLE
logging manifest type and don't log same error until next period in agent update.

### PR DESCRIPTION
**Pull request by nagworld9**
## Description

This PR includes the logging manifest type i.e. agent or extension and also, added period for logging same error in agent update.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests_e2e/pipeline/pipeline.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ parameters:
   - name: test_suites
     displayName: Test Suites
     type: string
-    default: agent_bvt
+    default: agent_bvt,agent_update
```
Comment (nagworld9) [inline tests_e2e/pipeline/pipeline.yml:12]: Added for daily monitoring agent update. After this merge, I'll schedule daily run for agent_update

File: azurelinuxagent/ga/agent_update.py [hunk 2]
```diff
@@ -170,15 +172,20 @@ def __get_all_agents_on_disk():
...
-            logger.error(msg_)
-        elif level == LogLevel.INFO:
+    def __log_event(self, level, msg_, success_=True):
+        if level == LogLevel.INFO:
             logger.info(msg_)
-        add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+            add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+        else:
+            msg_ += "[NOTE: Will not log the same error for the next 6 hours]"
+            if msg_ != self._last_warning or self._last_warning_time == datetime.datetime.min or datetime.datetime.now() >= self._last_warning_time + datetime.timedelta(hours=6):
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update.py:181]: does the message include any correlation id or anything that may be different in each invocation and make msg_ != self._last_warning be False?

Comment (nagworld9) [reply azurelinuxagent/ga/agent_update.py:181]: Great point, few messages include incarnation and may change on every goal state. Let me think about addressing that..

Comment (narrieta) [reply azurelinuxagent/ga/agent_update.py:181]: how often could this be logged if we didn't suppress?

Comment (nagworld9) [reply azurelinuxagent/ga/agent_update.py:181]: On every iteration, we check agent updates, so the frequency is 6 secs. But we have additional check of one hour window to allow updates. Before we compute this window there could be failures like missing agent family and we end up logging this error msg until any GS includes family section

Comment (nagworld9) [reply azurelinuxagent/ga/agent_update.py]: updated the logic and compare the substring if incarnation included in msg

File: azurelinuxagent/ga/agent_update.py [hunk 3]
```diff
@@ -170,15 +172,23 @@ def __get_all_agents_on_disk():
...
-            logger.error(msg_)
-        elif level == LogLevel.INFO:
+    def __log_event(self, level, msg_, success_=True):
+        if level == LogLevel.INFO:
             logger.info(msg_)
-        add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+            add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+        else:
+            msg_ += "[NOTE: Will not log the same error for the next 6 hours]"
+            # Incarnation may change if we get new goal state that would make whole string unique every time. So comparing only first partition if Incarnation included in msg
```
[the remark this answers is not in the document]
Comment (nagworld9) [reply azurelinuxagent/ga/agent_update.py]: Updated

**Review by narrieta — APPROVED** [review]

**Review by maddieford — APPROVED** [review]

**Pull request merged**